ZEngine/src/Texture2D.cpp

```diff
@@ -181,16 +181,18 @@ namespace ZEngine::Rendering::Textures
         uint32_t image_usage_attachment =
             (spec.Format == Specifications::ImageFormat::DEPTH_STENCIL_FROM_DEVICE) ? VK_IMAGE_USAGE_DEPTH_STENCIL_ATTACHMENT_BIT : VK_IMAGE_USAGE_COLOR_ATTACHMENT_BIT;
 
-        VkFormat image_format      = (spec.Format == Specifications::ImageFormat::DEPTH_STENCIL_FROM_DEVICE) ? Hardwares::VulkanDevice::FindDepthFormat()
-                                                                                                             : Specifications::ImageFormatMap[static_cast<uint32_t>(spec.Format)];
-        texture->m_image_2d_buffer = CreateRef<Buffers::Image2DBuffer>(
-            texture->m_width,
-            texture->m_height,
-            image_format,
-            VkImageUsageFlagBits(image_usage_attachment | transfert_bit | sampled_bit | storage_bit),
-            VkImageAspectFlagBits(image_aspect),
-            spec.LayerCount,
-            spec.IsCubemap ? VK_IMAGE_CREATE_CUBE_COMPATIBLE_BIT : 0);
+        VkFormat image_format = (spec.Format == Specifications::ImageFormat::DEPTH_STENCIL_FROM_DEVICE) ? Hardwares::VulkanDevice::FindDepthFormat()
+                                                                                                        : Specifications::ImageFormatMap[static_cast<uint32_t>(spec.Format)];
+        Specifications::Image2DBufferSpecification buffer_spec;
+        buffer_spec.Width         = texture->m_width;
+        buffer_spec.Height        = texture->m_height;
+        buffer_spec.ImageViewType = spec.IsCubemap ? Specifications::ImageViewType::TYPE_CUBE : Specifications::ImageViewType::TYPE_2D, buffer_spec.ImageFormat = image_format;
```

Comment:
we may want to introduce a BufferUsageType prop which will help to internally drive the needed bit/ type 
e.g buffer_spec.BufferUsageType   =  spec.IsCubemap ? (CUBEMAP | SINGLE_2D_IMAGE | SINGLE_3D_IMAGE | ARRAYOF_2D_IMAGE)